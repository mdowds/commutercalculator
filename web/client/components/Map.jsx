import React from 'react';

export default class Map extends React.Component {

    constructor(){
        super();
        this.state = { markers: [] };
    }

    componentDidMount() {
        const trafalgar = {lat: 51.507368, lng: -0.127811};

        this.map = new google.maps.Map(this.refs.map, {
            center: trafalgar,
            zoom: 11
        });

        this.setState({
            markers: [{
                position: trafalgar
            }]
        });
    }

    render() {

        this.state.markers.map((marker) => {
            return new google.maps.Marker({
                position: marker.position,
                map: this.map
            });
        });

        const mapStyle = {
            width: '100%',
            height: '100%'
        };

        return (
            <div ref="map" style={mapStyle}>I should be a map!</div>
        );
    }

}
