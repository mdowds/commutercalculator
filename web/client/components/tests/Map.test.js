import React from 'react';
import renderer from 'react-test-renderer';
import Map from '../Map.jsx';

test("Map renders the basic map container", () => {
    const map = renderer.create(
        <Map divRef={()=>{}}>Child content</Map>
    );
    let tree = map.toJSON();
    expect(tree).toMatchSnapshot();
});
