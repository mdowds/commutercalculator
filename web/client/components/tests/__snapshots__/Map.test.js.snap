// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Map renders the basic map container 1`] = `
<div
  style={
    Object {
      "height": "100%",
      "width": "100%",
    }
  }
>
  Map loading
  Child content
  ;
</div>
`;
